### PR TITLE
add dockerfile and scripts to make buildable and testable on jenkins

### PR DESCRIPTION
**Pull request by jfrazelle**
This doesn't work yet but the goal would be to have it run the integration tests w docker in docker.

you can try with:
```
docker build --rm --force-rm -f Dockerfile.test -t swarmtest .
docker run --rm -it --privileged swarmtest script/testall
```

I think you all will know better how the bats test needs the daemon started so I am opening this now, so hopefully you all will be able to spot the fix easily :)

### CODE REVIEW

**Comment by jfrazelle** [comment]
actually i think i might have it now

**Comment by jfrazelle** [comment]
nm idk

File: Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Dockerfile for testing swarm
+# How to use:
+#   docker build --rm --force-rm -f Dockerfile.test -t swarm-test .
+#   docker run --rm -it --privileged swarm-test script/testall
+FROM debian:jessie
```
Comment (aluzzardi) [inline Dockerfile.test]: How about starting from `aluzzardi/docker` so it comes pre-installed with docker and no need to install its dependencies?

File: Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Dockerfile for testing swarm
```
Comment (aluzzardi) [inline Dockerfile.test]: I think this Dockerfile could be located in `test/integration` (and simply named `Dockerfile`) and take care of only running integration tests (we are still using travis for unit)

File: Dockerfile.test [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ENV GOPATH /go
+RUN cd /usr/local/go/src && ./make.bash --no-clean 2>&1
+
+# install bats
+RUN cd /usr/local/src/ \
+    && git clone https://github.com/sstephenson/bats.git \
+    && cd bats \
+    && ./install.sh /usr/local
+
+# install go tools
```
Comment (aluzzardi) [inline Dockerfile.test]: These dependencies wouldn't be needed if we only run integration

File: script/dind [hunk 4]
```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
```
Comment (aluzzardi) [inline script/dind]: If we start from a docker image, this would be unnecessary

File: script/integration-test [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -e
+
+DEST=$1
+
+# subshell so that we can export PATH without breaking other things
+(
+	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start")
```
Comment (aluzzardi) [inline script/integration-test]: If I'm not mistaken, only one daemon per host can be run, right?

In that case, couldn't just we assume that docker is already running and available at `DOCKER_HOST` and not start it ourselves?

File: script/integration-test [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	
+	# we need to wrap up everything in between integration-daemon-start and
```
Comment (aluzzardi) [inline script/integration-test]: I **think** you could use `trap`: It calls some code before exiting the script (even if it was Ctrl-C'ed)

File: script/integration-test (hunk 5 above)
Comment (aluzzardi) [inline script/integration-test]: If this only runs integration tests (without starting the daemon), I think we could move it to `test/integration/run.sh` - not sure about that though.

Thoughts?

**Comment by aluzzardi** [comment]
Thanks a lot for this!

**Comment by aluzzardi** [comment]
Replaced by #696 (although heavily inspired by this).

**Pull request closed without merging**